homeassistant/components/jellyfin/browse_media.py

```diff
@@ -156,6 +158,68 @@ def fetch_items(
     ]
 
 
+async def search_items(
+    hass: HomeAssistant, client: JellyfinClient, user_id: str, query: SearchMediaQuery
+) -> list[BrowseMedia]:
+    """Search items in Jellyfin server."""
+    search_result: list[BrowseMedia] = []
+
+    items: list[dict[str, Any]] = []
+    # Search for items based on media filter classes (or all if none specified)
+    media_types: list[MediaClass] | list[None] = []
+    if query.media_filter_classes:
+        media_types = query.media_filter_classes
+    else:
+        media_types = [None]
+
+    for media_type in media_types:
+        items_dict: dict[str, Any] = await hass.async_add_executor_job(
+            partial(
+                client.jellyfin.search_media_items,
+                term=query.search_query,
+                media=media_type,
+                parent_id=query.media_content_id,
+            )
+        )
+        items.extend(items_dict.get("Items", []))
+
+    for item in items:
+        title = item["Name"]
+        thumbnail = get_artwork_url(client, item)
+        media: list[dict[str, Any]] | None = None
+        children: list[BrowseMedia] | None = None
+
+        if is_folder := item.get("IsFolder", False):
+            media = await hass.async_add_executor_job(
+                fetch_items, client, {"parentId": item["Id"], "fields": "childCount"}
+            )
+            media = media or []
+            children = await asyncio.gather(
+                *(
+                    item_payload(hass, client, user_id, media_item)
+                    for media_item in media
+                )
+            )
+
+        content_type: str = item["MediaType"]
+        response = BrowseMedia(
+            media_class=CONTAINER_TYPES_SPECIFIC_MEDIA_CLASS.get(
+                content_type, MediaClass.DIRECTORY
+            ),
+            media_content_id=item["Id"],
+            media_content_type=content_type,
+            title=title,
+            can_play=bool(content_type in PLAYABLE_MEDIA_TYPES),
+            can_expand=is_folder,
+            children=children,
```

Comment:
Why include all the children? I believe `children` should only be added to items that are being browsed, not for items that are inside the browsed item.

Reply:
Yeah, there is no UI that shows the children.